### DISCUSSION TITLE
Change generic types

### DISCUSSION THREAD

**Discussion by Unaxe** [Okampus/okampus#171] [Ideas]
A good practice is to don't use generic type for the infos like Email, phone number... 
I recommand to create an `email` type instead of the `string | null` type to avoid DataBase corruptions and any others problems

**Comment by Arno500** [comment]
It's a bit more complicated than that. Databases rarely support “rich” strings (like “emails”) and it's not their job to perform validation on them. Magically adding TypeScript types won't make the data “safe” nor perform any type of validation.
The embedded system SHOULD, however, support this kind of validation, but this is usually a system decoupled from the typing system.
